packages/embark/src/cmd/cmd_controller.js

```diff
@@ -955,5 +920,37 @@ async function compileAndDeploySmartContracts(engine) {
   let _contractsConfig = await engine.events.request2("config:contractsConfig");
   let contractsConfig = cloneDeep(_contractsConfig);
   let [contractsList, contractDependencies] = await engine.events.request2("contracts:build", contractsConfig, compiledContracts);
-  return await engine.events.request2("deployment:contracts:deploy", contractsList, contractDependencies);
+  return engine.events.request2("deployment:contracts:deploy", contractsList, contractDependencies);
+}
+
+async function setupCargoAndWatcher(engine) {
+  const cargo = async.cargo(async (tasks) => {
+    if (tasks.includes('contract') || tasks.includes('config')) {
+      try {
+        await compileAndDeploySmartContracts(engine);
+        return new Promise((resolve) => {
```

Comment:
I suggest instead of relying on `outputDone` we specifically wait for `await engine.events.request2('pipeline:generateAll'...)`, this way it's very explicity and we don't have to rely on an implicit event that might or might not be there

Reply:
Fixed